### DISCUSSION TITLE
Compatibilty of Hazelcast 4.x and Bucket4j 7.1+

### DISCUSSION THREAD

**Discussion by ayanahm** [vladimir-bukhtoyarov/bucket4j#247] [Q&A] [answered]
Hello there,

Beginning from 7.1 seems like Hazelcast 5.x is [expected](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-parent/pom.xml#L59) to be present in classspath. 
Can we still stick to Hazelcast 4.2 which was previously expected, or is it recommended to update to Hazelcast 5 now?

I couldn't find it in [documentation](https://bucket4j.com/7.3.0/toc.html#bucket4j-hazelcast) or in release notes, forgive me if I missed it.

Thank you.

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
Hello @ayanahm 

You are free to use Hazelcast 4.x, there are no requirements to switch to 5.x 

The compatibility matrix rules are simple: 
When Hazelcast maintainers broke backward compatibility, I just release yet another module [here](https://github.com/vladimir-bukhtoyarov/bucket4j/tree/master/bucket4j-hazelcast-all). For example if new version Hazelcast 6.0 will not be compatible with Bucket4j, I  will add new module bucket4j-hazelcast-5 and ```bucket4j-hazelcast``` will depend from fresh Hazelcast 6.x.
